core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/EmailAction.java

```diff
@@ -0,0 +1,191 @@
+      authenticator = new Authenticator() {
+        @Override
+        public PasswordAuthentication getPasswordAuthentication() {
+          return new PasswordAuthentication(config.username, config.password);
+        }
+      };
+    }
+    if ("SMTPS".equalsIgnoreCase(config.protocol)) {
+      javaMailProperties.put("mail.smtp.ssl.enable", true);
+    }
+    if ("TLS".equalsIgnoreCase(config.protocol)) {
+      javaMailProperties.put("mail.smtp.starttls.enable", true);
+    }
+
+    Session session = Session.getInstance(javaMailProperties, authenticator);
+    session.setDebug(true);
+
+    try {
+      Message msg = new MimeMessage(session);
+      msg.setFrom(new InternetAddress(config.senderEmailAddress));
+      msg.addRecipient(Message.RecipientType.TO, new InternetAddress(config.recipientEmailAddress));
+      msg.setSubject(config.subject);
+      WorkflowToken token = context.getToken();
+      String message = config.includeWorkflowToken ?
+        config.message + "\nUSER Workflow Tokens:\n" + token.getAll(WorkflowToken.Scope.USER)
+          + "\nSYSTEM Workflow Tokens:\n" + token.getAll(WorkflowToken.Scope.SYSTEM) :
+        config.message;
+      msg.setText(message);
+
+      // need this because Session will use the context classloader to instantiate an object.
+      // the context classloader here is the etl application's classloader and not this class' classloader.
+      ClassLoader oldClassLoader = Thread.currentThread().getContextClassLoader();
+      Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+      try {
+        Transport transport = session.getTransport(config.protocol);
+        transport.connect(config.host, config.port, config.username, config.password);
+        try {
+          transport.sendMessage(msg, msg.getAllRecipients());
+        } finally {
+          transport.close();
+        }
+      } finally {
+        Thread.currentThread().setContextClassLoader(oldClassLoader);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Error sending email: ", e);
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    if (Strings.isNullOrEmpty(config.username) ^ Strings.isNullOrEmpty(config.password)) {
+      throw new IllegalArgumentException(
+        String.format("You must either set both username and password " +
+                        "or neither username nor password. " +
+                        "Currently, they are username: \'%s\', and password: \'%s\'",
+                      config.username, config.password));
+    }
+
+    try {
+      new InternetAddress(config.senderEmailAddress);
+    } catch (AddressException e) {
+      throw new IllegalArgumentException(
+        String.format("%s is an invalid sender email address. Reason: %s",
+                      config.senderEmailAddress, e.getMessage()));
+    }
+
+    try {
+      new InternetAddress(config.recipientEmailAddress);
```

Comment:
hm though it may be a good idea to support multiple recipients.  Think I will change it so that is supported